src/self_signed_certificates.py

```diff
@@ -68,23 +68,21 @@ def generate_certificate(
     issuer = x509.load_pem_x509_certificate(ca).issuer
     private_key = serialization.load_pem_private_key(ca_key, password=ca_key_password)
 
-    certificate_builder = (
-        x509.CertificateBuilder()
-        .subject_name(subject)
-        .issuer_name(issuer)
-        .public_key(csr_object.public_key())
-        .serial_number(x509.random_serial_number())
-        .not_valid_before(datetime.datetime.utcnow())
-        .not_valid_after(datetime.datetime.utcnow() + datetime.timedelta(days=validity))
+    certificate_builder = x509.CertificateBuilder(
+        subject_name=subject,
+        issuer_name=issuer,
+        public_key=csr_object.public_key(),
+        serial_number=x509.random_serial_number(),
+        not_valid_before=datetime.datetime.utcnow(),
+        not_valid_after=datetime.datetime.utcnow() + datetime.timedelta(days=validity),
     )
 
-    if alt_names:
-        names = [x509.DNSName(n) for n in alt_names]
+    for extension in csr_object.extensions:
         certificate_builder = certificate_builder.add_extension(
-            x509.SubjectAlternativeName(names),
-            critical=False,
+            extension.value,
+            critical=extension.critical,
         )
-    certificate_builder._version = x509.Version.v1
+    certificate_builder._version = x509.Version.v3
```

Comment:
Is this change related to fixing the SAN issue?

Reply:
The extensions we are setting are of type: x509.v3, it is then necessary to set the version to the same. Otherwise, the loading of extensions fails with javax.net.ssl.SSLProtocolException: no more data allowed for version 1 certificate.